### PR TITLE
p2p example

### PR DESCRIPTION
**Pull request by loocapro**
Added an example that establishes p2p sessions with polygon peers.

Credits: https://blog.merkle.io/blog/fastest-transaction-network-eth-polygon-bsc

### CODE REVIEW

File: examples/polygon-p2p/src/chain_cfg.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+use reth_primitives::{
+    b256, BaseFeeParams, Chain, ChainSpec, ForkCondition, ForkTimestamps, Hardfork, Head,
+    NodeRecord, B256,
+};
+
+use std::{collections::BTreeMap, sync::Arc};
+
+const SHANGAI_BLOCK: u64 = 50523000;
+
+pub(crate) fn chain_spec() -> Arc<ChainSpec> {
```
Comment (mattsse) [inline examples/polygon-p2p/src/chain_cfg.rs:10]:
```suggestion
pub(crate) fn polygon_chain_spec() -> Arc<ChainSpec> {
```

File: examples/polygon-p2p/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+//! ```
+//!
+//! This launch the regular reth node overriding the engine api payload builder with our custom.
+//!
+//! Credits to: <https://blog.merkle.io/blog/fastest-transaction-network-eth-polygon-bsc>
+use chain_cfg::{boot_nodes, chain_spec, head};
+use reth_network::{
+    config::NetworkMode, NetworkConfig, NetworkEvent, NetworkEvents, NetworkManager,
+};
+
```
Comment (mattsse) [inline examples/polygon-p2p/src/main.rs:16]:
```suggestion
```

File: examples/polygon-p2p/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    // The local address we want to bind to
+    let local_addr = SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), 30303);
+
+    // The network configuration
+    let net_cfg = NetworkConfig::builder(secret_key)
+        .chain_spec(chain_spec())
+        .set_head(head())
+        .network_mode(NetworkMode::Work)
+        .listener_addr(local_addr)
+        .build(NoopProvider::default());
```
Comment (mattsse) [inline examples/polygon-p2p/src/main.rs:54]: I think we want a helper function for this in a followup `build_with_noop_provider`

https://github.com/paradigmxyz/reth/blob/30aa3e6792701a8719a49995d79a0525bfc33881/crates/net/network/src/config.rs#L421-L421

the provider in p2p isn't that useful to begin with

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, a few nits

I'd also like to setup the txs manager, see other example so we can get polygon txs

**Comment by emhane** [comment]
reason for closing this @loocapro ?

**Comment by loocapro** [comment]
> reason for closing this @loocapro ?

its already merged here: https://github.com/paradigmxyz/reth/pull/6393

**Pull request closed without merging**